sno-setup.sh

```diff
@@ -39,15 +39,15 @@ else
   sshKeyFile=~/.ssh/id_rsa.pub
 fi
 
-if $(sudo virsh list --all | grep "\s${hostname}\s" &> /dev/null); then
-  echo "Found an already existing mini-agent instance, please run sno-cleanup.sh to remove it"
+if [ -d "${assets_dir}" ] || sudo virsh list --all --name | grep -q "${hostname}"; then
```

Comment:
I like the approach, but then also the check about the presence or not of the network will make this more robust (the user may have deleted the VM, but not the network)

Reply:
Good catch, will add!